lib/getoptlong.rb

```diff
@@ -6,83 +6,367 @@
 #
 # You may redistribute and/or modify this library under the same license
 # terms as Ruby.
+
+# \Class \GetoptLong provides command-line parsing both for options
+# and for regular arguments.
 #
-# See GetoptLong for documentation.
+# Using \GetoptLong, you can define command-line options for your program.
+# The program can then capture and respond to whatever options
+# are included in the command that executes the program.
 #
-# Additional documents and the latest version of `getoptlong.rb' can be
-# found at http://www.sra.co.jp/people/m-kasahr/ruby/getoptlong/
-
-# The GetoptLong class allows you to parse command line options similarly to
-# the GNU getopt_long() C library call. Note, however, that GetoptLong is a
-# pure Ruby implementation.
+# A simple example: file <tt>simple.rb</tt>:
+#
+#   :include: ../doc/simple.rb
```

Comment:
How does this including work for usage as a default gem in Ruby?  I'm guessing it wouldn't work, since we don't want to have `simple.rb` in `doc` in the Ruby repository.  Possibly we could use a `doc/getoptlong` directory instead.  Not sure if that is desired.  Hopefully @hsbt can provide input.

Reply:
Will wait for direction on this one.

Reply:
Have moved included files to doc/getoptlong, which is consistent with other default gems' usage.  Also (the horror!), I did not realize that directory doc was in the .gitignore file, and so only now are the .rdoc files getting into this PR.